### PR TITLE
support funnel flatten the top to represent none zero value

### PR DESCRIPTION
**Pull request by Chellhooo**
## Brief Information

1. support funnel flatten the top to represent none zero value
    image
    
    copied by https://github.com/apache/echarts/pull/17461
    doc: https://github.com/apache/echarts-doc/pull/281
2. publish
test publish job: https://github.com/appflow-ai/echarts/actions/runs/4370931308/jobs/7646315202

### CODE REVIEW

File: test/runTest/util.js [hunk 1]
```diff
@@ -102,12 +102,12 @@ module.exports.fetchVersions = function (isNighlty) {
     return new Promise((resolve, reject) => {
         https.get(
             isNighlty
```
Comment (Xhhhhhh) [inline test/runTest/util.js]: It seems we do not use `nightly`.

File: package.json [hunk 2]
```diff
@@ -106,5 +106,8 @@
     "terser": "^5.16.1",
     "ts-jest": "^26.4.3",
     "typescript": "4.4.3"
+  },
+   "publishConfig": {
```
Comment (Xhhhhhh) [inline package.json]: Does overlap with .npmrc?

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -22,7 +23,7 @@ jobs:
       - name: Use Node.js ${{ matrix.node-version }}
         uses: actions/setup-node@v3
         with:
-          registry-url: https://registry.npmjs.org/
+          registry-url: 'https://npm.pkg.github.com'
```
Comment (Xhhhhhh) [inline .github/workflows/publish.yml]: 'https://npm.pkg.github.com' -> https://npm.pkg.github.com

File: src/chart/funnel/FunnelSeries.ts [hunk 4]
```diff
@@ -48,6 +48,10 @@ type FunnelLabelOption = Omit<SeriesLabelOption, 'position'> & {
         | 'outer' | 'inner' | 'center' | 'rightTop' | 'rightBottom' | 'leftTop' | 'leftBottom'
 };
 
+type FunnelRateLabelOption = Omit<SeriesLabelOption, 'position'> & {
```
Comment (Xhhhhhh) [inline src/chart/funnel/FunnelSeries.ts:51]: This commit should be merge in "apache/echarts" repository. why need add this change manually ?  why not pull new updates from original GitHub repository into forked GitHub repository?

**Comment by Xhhhhhh** [comment]
If this repository is managed by us, please add https://implustech.atlassian.net/wiki/spaces/TECH/pages/1958215698/How+we+use+github#Pull-Request-Template firstly.

File: test/runTest/util.js (hunk 1 above)
Comment (Chellhooo) [reply test/runTest/util.js]: I will change nothing here

File: src/chart/funnel/FunnelSeries.ts (hunk 4 above)
Comment (Chellhooo) [reply src/chart/funnel/FunnelSeries.ts:51]: > This commit should be merge in "apache/echarts" repository

no, it is in  `NextFunnel` branch

File: .github/workflows/publish.yml (hunk 3 above)
Comment (Chellhooo) [reply .github/workflows/publish.yml]: updated

File: package.json (hunk 2 above)
Comment (Chellhooo) [reply package.json]: removed

**Comment by Chellhooo** [comment]
@Xhhhhhh please check https://github.com/appflow-ai/echarts/pull/2

File: test/runTest/util.js (hunk 1 above)
Comment (Xhhhhhh) [reply test/runTest/util.js]: https://github.com/appflow-ai/echarts/blob/master/test/runTest/server.js#L234

I saw you delete the yml config of punish nightly version but It depend on nightly version somewhere?

Comment (Chellhooo) [reply test/runTest/util.js]: actually I just don't to run night job, it do not mean it need to be removed

File: src/chart/funnel/FunnelSeries.ts (hunk 4 above)
Comment (Xhhhhhh) [reply src/chart/funnel/FunnelSeries.ts:51]: > > This commit should be merge in "apache/echarts" repository
> 
> no, it is in `NextFunnel` branch

please review this change 
@zzn01

**Review by zzn01 — APPROVED** [review]

**Pull request merged**